worlds/blasphemous/__init__.py

```diff
@@ -61,102 +69,105 @@ def get_filler_item_name(self) -> str:
         return self.multiworld.random.choice(tears_set)
 
 
-    def create_items(self):
-        placed_items = []
-
-        placed_items.extend(Vanilla.unrandomized_dict.values())
-
-        if not self.multiworld.reliquary_shuffle[self.player]:
-            placed_items.extend(reliquary_set)
-        elif self.multiworld.reliquary_shuffle[self.player]:
-            placed_items.append("Tears of Atonement (250)")
-            placed_items.append("Tears of Atonement (300)")
-            placed_items.append("Tears of Atonement (500)")
-
-        if not self.multiworld.cherub_shuffle[self.player]:
-            for i in range(38):
-                placed_items.append("Child of Moonlight")
-
-        if not self.multiworld.life_shuffle[self.player]:
-            for i in range(6):
-                placed_items.append("Life Upgrade")
-
-        if not self.multiworld.fervour_shuffle[self.player]:
-            for i in range(6):
-                placed_items.append("Fervour Upgrade")
-
-        if not self.multiworld.sword_shuffle[self.player]:
-            for i in range(7):
-                placed_items.append("Mea Culpa Upgrade")
-
-        if not self.multiworld.blessing_shuffle[self.player]:
-            placed_items.extend(Vanilla.blessing_dict.values())
-
-        if not self.multiworld.dungeon_shuffle[self.player]:
-            placed_items.extend(Vanilla.dungeon_dict.values())
-
-        if not self.multiworld.tirso_shuffle[self.player]:
-            placed_items.extend(Vanilla.tirso_dict.values())
-
-        if not self.multiworld.miriam_shuffle[self.player]:
-            placed_items.append("Cantina of the Blue Rose")
+    def generate_early(self):
+        world = self.multiworld
+        player = self.player
 
-        if not self.multiworld.redento_shuffle[self.player]:
-            placed_items.extend(Vanilla.redento_dict.values())
+        if world.starting_location[player].value == 6 and world.difficulty[player].value < 2:
+            raise Exception(f'[Blasphemous - "{world.get_player_name(player)}"] {world.starting_location[player]}'
+                            ' cannot be chosen if Difficulty is lower than Hard.')
```

Comment:
Incompatible settings raising Exceptions is usually not appreciated because people use mystery yamls in big asyncs, which can cause a lot of problems. Some AP core devs tend to prefer a solution that "fixes" the settings in some way. I would investigate whether that's possible if you haven't already

If it's not possible (Like I determined in Witness, which still has an Exception), I would at least suggest setting the process in motion for getting a trigger added to the filler slot yaml for the big AP asyncs that avoids this issue. I don't remember exactly how to do that, but you should be able to ask somewhere, probably #async-general.

Reply:
Besides the one that checks for difficulty, the exceptions have to be there because you would either be softlocked instantly, or the first item you pick up would need to be the same item every single time before you could do anything else. I will look into getting a trigger added now, though.